### ISSUE TITLE
HTML spaces removed completely on line endings

### ISSUE THREAD

**Issue by nigelboor** [Josee9988/MinifyAll#16]
When minifying HTML file some of the words are combined.

Eg:
This...
***
                                                        <p>The role will involve working within Gloucestershire&rsquo;s
                                                            countywide Continuing Health Care Team. You will be
                                                            completing CHC assessments and taking a vital role in the
                                                            multi-disciplinary team to complete the Decision Support
                                                            Tool (DST). This will include making decisions around
                                                            eligibility of a primary health need. You will work with
                                                            colleagues within the CHC team to ensure that a seamless
                                                            service is provided to all individuals who may become
                                                            eligible for CHC, have become ineligible for CHC or are CHC
                                                            funded. You will be involved in supporting the development
                                                            of partnership CHC practice by sharing CHC information with
                                                            locality champions for cascading to locality staff under the
                                                            direction of the CHC operational lead.</p>
***

Is minified to...
***
<p>The role will involve working within Gloucestershire&rsquo;scountywide Continuing Health Care Team. You will becompleting CHC assessments and taking a vital role in themulti-disciplinary team to complete the Decision SupportTool (DST). This will include making decisions aroundeligibility of a primary health need. You will work withcolleagues within the CHC team to ensure that a seamlessservice is provided to all individuals who may becomeeligible for CHC, have become ineligible for CHC or are CHCfunded. You will be involved in supporting the developmentof partnership CHC practice by sharing CHC information withlocality champions for cascading to locality staff under thedirection of the CHC operational lead.</p>

***

VSCode is set to use spaces for indenting

**Comment by Josee9988** [comment]
Hi @nigelboor , when I minify your code this is what it gives me:
```html
<p>The role will involve working within Gloucestershire&rsquo;scountywide Continuing Health Care Team. You will becompleting CHC assessments and taking a vital role in themulti-disciplinary team to complete the Decision SupportTool (DST). This will include making decisions aroundeligibility of a primary health need. You will work withcolleagues within the CHC team to ensure that a seamlessservice is provided to all individuals who may becomeeligible for CHC, have become ineligible for CHC or are CHCfunded. You will be involved in supporting the developmentof partnership CHC practice by sharing CHC information withlocality champions for cascading to locality staff under thedirection of the CHC operational lead.</p>
```

I see the issue, and I'll try to fix this as soon as possible, thanks for letting me know!

**Comment by Josee9988** [comment]
Hey, I've found there is not an existent issue in the spaces, but it is giving some bugs because your text is separated in multiple lines, and the extension tries to avoid multiple lines by removing them and creating just one line.
If I let a space when we found a linebreak it will create multiple spaces between HTML labels and also within words.

The problem occurs because for example, the result: 
```html
locality champions for cascading to locality staff under the
                                                        direction
```

will give 
```html
thedirection
```

because after **the** it literally finds a linebreak and not space, so the extension could not imagine that this linebreak is a space, so it is not an extension bug, it is a problem with the code you gave, that is comprehensible for you but it is not for the extension.

To sum up all the mess I said, the extension detects spaces correctly but can not understand if linebreaks mean a space to you, this could be solved by adding a space after the word you want to have space, even if there is a linebreak after it.

Thanks for the interest!

**Comment by nigelboor** [comment]
The solution might be to first convert all line feeds to spaces then minify.

There’s no requirement in text files to terminate a line with space+return so this is something that the extension should handle.
It’s a setting in VSCode
"editor.trimAutoWhitespace": true



--
Nigel Boor

**Comment by Josee9988** [comment]
I'll take a deeper look in how to handle it. Stay tuned.

**Comment by Josee9988** [comment]
It is **fixed** in the new version **2.1.1**.
Download it and test it out, if there is any other problem or the original one wasn't solved as you expected to, feel free to keep posting issues or simply telling me.

Thanks for trying to make this extension better, your help is highly appreciated!

**Comment by nigelboor** [comment]
That fixes the problem, thanks.

I can see one remaining issue: the extension still reduces “</a> “ to “</a>” – it should preserve the space.

I’ve also checked two other inline elements <strong> and <sup>, and similarly “</strong> “ is reduced to “</strong>” and “</sup> “ to “</sup>”.

Other inline elements will have similar issues when trailing spaces are removed although which other HTML elements should have preserved trailing space would need evaluation.



--
Nigel Boor

**Comment by Josee9988** [comment]
Ow, you are completely right, but I'm afraid to say this won't be fixed anytime soon, because I'm doing it with regex and there is no way to distinct if the "> " that I'm looking is from a strong label or not. :sleepy:

**Comment by nigelboor** [comment]
Good luck & I hope you crack it.

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________
